### ISSUE TITLE
Wayland icon instead of Ente Auth's icon in KDE taskbar.

### ISSUE THREAD

**Issue by techiepk** [ente-io/ente#4329]
### Description

It can be observed that the KDE taskbar doesn't display the Ente Auth icon in the taskbar, and instead shows the wayland placeholder icon. I can safely asume that the same goes on Gnome, with their placeholder icon. that makes switching apps tedious, since multiple apps lack proper icons. 

Attachment for reference: 
![Screenshot_20241206_213909](https://github.com/user-attachments/assets/c8594b57-c9fb-4e1a-822a-f6309931c9a8)

I'm using the RPM package on Nobara Linux at the moment, but observed the same behaviour on the AUR package on Cachy OS.

### Version

4.1.5

### What product are you using?

Ente Auth

### What platform are you using?

Desktop - Linux

**Comment by namher-sec** [comment]
Yes, I'm having the same issue in Fedora 41 KDE. This happened after they changed the package from ente_auth to enteauth.

**Comment by znarfm** [comment]
Hello, just wanted to gently check in on the status of this issue regarding the Wayland icon display in KDE taskbar. Would appreciate any update on this since it is still occurring on my end. Thank you!

**Comment by jassielof** [comment]
This issue also affects the task switcher icon in Kubuntu 25.04

**Comment by jassielof** [comment]
The issue seems to be related to the app ID resolution, at least in KDE 6.4 setting a special window rule to the desktop file `enteauth` (installed via `deb` package), solves the issue. Also renaming the desktop file to `io.ente.auth.desktop` solves it.
It's basically an app ID mismatch issue.

**Comment by ua741** [comment]
This should should be fixed in the latest release. Please comment here if you are still facing this issue.
